### PR TITLE
[SAM] Meikyo bugifx

### PR DESCRIPTION
**Pull request by Rjb99**
Bugfix for Meikyo because apparently, Samurais are unable to bend space time, my bad.

Short tl;dr EoF cursed, can return possible GCDs that exceed 3. Hard capping it to 3.

### CODE REVIEW

File: src/parser/jobs/sam/modules/Meikyo.tsx [hunk 1]
```diff
@@ -94,6 +94,9 @@ export class Meikyo extends BuffWindow {
 			const fightTimeRemaining = (this.parser.pull.timestamp + this.parser.pull.duration) - window.start
 			const possibleGCDs = Math.floor(fightTimeRemaining / SAM_BASE_GCD_SPEED_BUFFED)
 
+			if (possibleGCDs > SEN_GCDS) { //No Samurai, you can not slash so fast you bend space-time and get 2 GCDs for the price of 1
+				return 0
+			}
 			return possibleGCDs - defaultEoFValue
```
Comment (ayyaruq) [inline src/parser/jobs/sam/modules/Meikyo.tsx]: You return 0 later, so I think you can simplify this. If you return the `possibleGCDs - defaultEoFValue` with an inverted condition, you can keep the later `return 0` and make this simpler.

File: src/parser/jobs/sam/changelog.tsx [hunk 2]
```diff
@@ -8,6 +8,10 @@ export const changelog = [
 	// 	contrubutors: [CONTRIBUTORS.YOU],
 	// },
 	{
+		date: new Date('2022-1-28'),
```
Comment (ayyaruq) [inline src/parser/jobs/sam/changelog.tsx]:
```suggestion
		date: new Date('2022-01-28'),
```

**Review by ayyaruq — CHANGES_REQUESTED** [review]
Not requesting code changes, just noting the condition could be simplified. Changelog needs fixing tho. Surprised the older entry parses sanely since it has the same issue.

File: src/parser/jobs/sam/modules/Meikyo.tsx (hunk 1 above)
Comment (hintxiv) [reply src/parser/jobs/sam/modules/Meikyo.tsx:100]:
```suggestion
                        return Math.min(SEN_GCDs, possibleGCDs - defaultEoFValue)
```

**Comment by Rjb99** [comment]
@ayyaruq requested changes made

**Review by AzariahAeroborn — APPROVED** [review]

**Review by ayyaruq — APPROVED** [review]

**Pull request merged**